src/playdate/bindings/scoreboards.nim

```diff
@@ -2,45 +2,48 @@
 
 import utils
 
-type PDScore* {.importc: "PDScore", header: "pd_api_scoreboards.h", bycopy.} = object
-    rank* {.importc: "rank".}: uint32
-    value* {.importc: "value".}: uint32
+type PDScoreRaw* {.importc: "PDScore", header: "pd_api.h", bycopy.} = object
+    rank* {.importc: "rank".}: cuint
+    value* {.importc: "value".}: cuint
     player* {.importc: "player".}: cstring
 
-type PDScoresList* {.importc: "PDScoresList", header: "pd_api_scoreboards.h", bycopy.} = object
+type PDScorePtr* = ptr PDScoreRaw
+# type PDScore* = ref PDScoreRaw
+
+type PDScoresListRaw* {.importc: "PDScoresList", header: "pd_api.h", bycopy.} = object
     boardID* {.importc: "boardID".}: cstring
     count* {.importc: "count".}: cuint
-    lastUpdated* {.importc: "lastUpdated".}: uint32
+    lastUpdated* {.importc: "lastUpdated".}: cuint
     playerIncluded* {.importc: "playerIncluded".}: cint
     limit* {.importc: "limit".}: cuint
-    scores* {.importc: "scores".}: ptr PDScore
+    scores* {.importc: "scores".}: ptr PDScoreRaw
 
-type PDBoard* {.importc: "PDBoard", header: "pd_api_scoreboards.h", bycopy.} = object
+type PDBoardRaw* {.importc: "PDBoard", header: "pd_api.h", bycopy.} = object
     boardID* {.importc: "boardID".}: cstring
     name* {.importc: "name".}: cstring
 
-type PDBoardsList* {.importc: "PDBoardsList", header: "pd_api_scoreboards.h", bycopy.} = object
+type PDBoardsListRaw* {.importc: "PDBoardsList", header: "pd_api.h", bycopy.} = object
     count* {.importc: "count".}: cuint
-    lastUpdated* {.importc: "lastUpdated".}: uint32
-    boards* {.importc: "boards".}: ptr PDBoard
+    lastUpdated* {.importc: "lastUpdated".}: cuint
+    boards* {.importc: "boards".}: ptr PDBoardRaw
 
-type AddScoreCallback* = proc (score: ptr PDScore; errorMessage: cstring) {.cdecl.}
-type PersonalBestCallback* = proc (score: ptr PDScore; errorMessage: cstring) {.cdecl.}
-type BoardsListCallback* = proc (boards: ptr PDBoardsList; errorMessage: cstring) {.cdecl.}
-type ScoresCallback* = proc (scores: ptr PDScoresList; errorMessage: cstring) {.cdecl.}
+type PersonalBestCallbackRaw* {.importc: "PersonalBestCallback", header: "pd_api.h".} = proc (score: PDScorePtr; errorMessage: cstring) {.cdecl.}
+type AddScoreCallbackRaw* {.importc: "AddScoreCallback", header: "pd_api.h".} = proc (score: PDScorePtr; errorMessage: cstring) {.cdecl.}
+# type BoardsListCallbackRaw* = proc (boards: ptr PDBoardsListRaw; errorMessage: cstring) {.cdecl.}
+# type ScoresCallbackRaw* = proc (scores: ptr PDScoresListRaw; errorMessage: cstring) {.cdecl.}
```

Comment:
Why comment these out instead of deleting?

Reply:
Will be re added before merge, this is a draft pr. I'm first implementing some functions and then gathering feedback from you guys.